params/config.go

```diff
@@ -283,10 +293,47 @@ func (c *ChainConfig) CheckCompatible(newcfg *ChainConfig, height uint64) *Confi
 	return lasterr
 }
 
+// CheckConfigForkOrder checks that we don't "skip" any forks, geth isn't pluggable enough
+// to guarantee that forks can be implemented in a different order than on official networks
+func (c *ChainConfig) CheckConfigForkOrder() error {
+	type fork struct {
+		name     string
+		block    *big.Int
+		optional bool // if true, the fork may be nil and next fork is still allowed
+	}
+	var lastFork fork
+	for _, cur := range []fork{
+		{name: "istanbulBlock", block: c.IstanbulCompatibleBlock},
+		{name: "londonBlock", block: c.LondonCompatibleBlock},
+	} {
+		if lastFork.name != "" {
```

Comment:
Since var `lastfork` is declared on line 304 and nothing is assigned to that variable, I wonder how `lastfork.name !=""` is possible on line 309.

Reply:
The assignment is done at line 324. `lastFork` is used to memorize the forkName of previous loop :).